File: fix_keys.py
import re

with open('src/components/use-case/SingleAssessmentPageComponent/Assessments/CoreArea3_Zakat/CRITERIA_OPTION_TEXT.ts', 'r') as f:
    content = f.read()

content = re.sub(r'"P1\.(\d+)":', r'"S2.\1":', content)
content = re.sub(r'"E1\.(\d+)":', r'"S3.\1":', content)
content = re.sub(r'"T1\.(\d+)":', r'"S4.\1":', content)
content = re.sub(r'"C1\.(\d+)":', r'"S5.\1":', content)

with open('src/components/use-case/SingleAssessmentPageComponent/Assessments/CoreArea3_Zakat/CRITERIA_OPTION_TEXT.ts', 'w') as f:
    f.write(content)

File: fix_keys.js
const fs = require('fs');
let content = fs.readFileSync('src/components/use-case/SingleAssessmentPageComponent/Assessments/CoreArea3_Zakat/CRITERIA_OPTION_TEXT.ts', 'utf8');

// Replace P1.x with S2.x
content = content.replace(/"P1\.(\d+)":/g, '"S2.$1":');
// Replace E1.x with S3.x
content = content.replace(/"E1\.(\d+)":/g, '"S3.$1":');
// Replace T1.x with S4.x
content = content.replace(/"T1\.(\d+)":/g, '"S4.$1":');
// Replace C1.x with S5.x
content = content.replace(/"C1\.(\d+)":/g, '"S5.$1":');

fs.writeFileSync('src/components/use-case/SingleAssessmentPageComponent/Assessments/CoreArea3_Zakat/CRITERIA_OPTION_TEXT.ts', content);
